### ISSUE TITLE
AEMM Android not building for release

### ISSUE THREAD

**Issue by Robert-Fairley** [adobe-marketing-cloud-mobile/aemmobile#65]
Hi,
I thought I would post this here, as I am not sure if my thread on the Adobe forums is visible any longer. I was pointed to this repo as the possible source of the issue.

If you don't mind, for a full description, conversation, and screen caps, please see: 

[https://forums.adobe.com/message/9143010](https://forums.adobe.com/message/9143010)

_This began occurring in mid-November_

---

## A description of the problem (as posted originally):

Hi, I am trying to build custom Cordova plugins into an AEM mobile application and while my build and deployment works perfectly fine on iOS, I am having endless trouble getting it to work on Android at all.
 
I must clarify, I didn't have this problem in my first build, but shortly after adding the aemm Cordova plugins, my builds on Android ceased working. Further, the build process goes perfectly. I haven't been thrown any errors except for the one listed in the subject.
The error occurs when I attempt to run the app in the emulator for testing purposes, or when I try to upload the APK as a custom shell to rebuild the AEM application.
 
I haven't been able to find any related information yet, and including custom plugins is crucial to a module we are trying to include at our clients' request.
 
At first I suspected something may be wrong with my build so I created a fresh project and did not include any plugins and received the same result.
 
Next, thinking that something may have been corrupted I removed all traces of Android Studio and the AEM tool, all SDK's, and updated Node and NPM to be sure. I then reinstalled and recreated a project, strictly following the instructions in the docs and received the same result.
 
As a last resort I attempted to do a fresh build on another Mac. The other computer had never had any related tools installed previously, so installed per Adobe's instructions again. And again without, followed by with, the plugins required. Both times failed with the same error.
 
After running `aemm run android` I only receive (in terminal):
 
```
Success
starting app server...
Use Ctrl-C to exit
listening on 10.10.114.154:3000
listening on 10.10.118.23:3000
Starting: Intent { cmp=com.adobe.dps.viewer/.collectionview.CollectionActivity (has extras) }
200 '/articles.json'
200 '/articles.json'
```
 
TL;DR: My Android builds are not working and throwing the error in the subject. I had it working prior to installing the standard bunch of aemm plugins, and it works on iOS without a problem. I need help getting this working as we require a few custom plugins for a module being included in HTML articles.

---

## In response to a request for commands run, verbatim:

Up until I ran the following everything worked:

 ```
aemm plugin add aemm-plugin-application aemm-plugin-user aemm-plugin-context aemm-plugin-device aemm-plugin-inappbrowser aemm-plugin-fullscreen-video aemm-plugin-navto aemm-plugin-html-contract
aemm plugin add https://github.com/adobe-marketing-cloud-mobile/cordova-plugin-wkwebview-engine.git#2.0.0
aemm plugin add cordova-plugin-camera cordova-plugin-compat cordova-plugin-contacts cordova-plugin-device cordova-plugin-device-motion cordova-plugin-device-orientation cordova-plugin-dialogs cordova-plugin-file cordova-plugin-file-transfer cordova-plugin-geolocation cordova-plugin-globalization cordova-plugin-media cordova-plugin-media-capture cordova-plugin-network-information cordova-plugin-vibration
 ```

It was after this that I ran the these commands to find it failing:
 
```
aemm build android
aemm run android
```

Since those plugins, I tried removing them to no avail. I tried starting fresh projects and adding no plugins to no avail, as such:
 
```
aemm project create [project]
aemm platform add android
aemm build android
aemm run android
```

Each command ran without error, on both systems I tested. It would work prior to adding the standard plugin set (I even used custom plugins in the first build prior to adding the standard set), and would fail afterward. Then no matter what I do I haven't been able to see any change in the result.

---

_Edit:_ After some support from one of your contributors, I was able to get an Android build functioning on the emulator only. All build attempts with the produced "Custom Shell APK" fail.

Many thanks for any help/consideration!
